### PR TITLE
feat: enable local proxy configuration in maven plugin configuration

### PR DESCRIPTION
**Pull request by mischa-n**
## Description of Change

CICD-SEC-03 recommends:
1. ... packages are pulled through an internal proxy rather than directly from the internet ...
2. ... pull packages from internal repositories ...

We have solved (2), trust our developers, they trust the developers of the third party libraries in use, and those trust the developers of dependent fourth party libraries ...
Nevertheless, we would like to go one step further and completely disable access to the Internet in our CI/CD pipeline.
Therefore, we need to configure a proxy for the dependency-check-maven plugin.

Currently, you can define a proxy to use in the dependency-check-maven plugin via the Maven settings or the system property `http.proxyHost`.
However, in both cases, the configured proxy is used not only for loading data for analysis, but potentially for dependency resolution as well:
* https://maven.apache.org/guides/mini/guide-proxies.html "... configure a proxy to use for some or all of your HTTP requests with Maven ..."
* https://www.baeldung.com/maven-behind-proxy

With the proposed change, a local proxy can be configured for the dependency-check-maven plugin, which is then used exclusively for loading the analysis data:

```
<configuration>
	...
	<proxy>
		<host>myproxy</host>
		<port>1234</port>
	</proxy>
	...
</configuration>
```

## Have test cases been added to cover the new functionality?

*no*

### CODE REVIEW

File: maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java [hunk 1]
```diff
@@ -2197,6 +2200,14 @@ protected void populateSettings() {
                 settings.setString(Settings.KEYS.PROXY_NON_PROXY_HOSTS, System.getProperty("http.nonProxyHosts"));
             }
         }
+        // or use configured <proxy>
+        else if ( this.proxy != null && this.proxy.host != null) {
+            settings.setString(Settings.KEYS.PROXY_SERVER, this.proxy.host);
```
Comment (jeremylong) [inline maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java:2205]: Wouldn't you also want to be able to store these values in the settings.xml so you don't have to store the credentials in clear text? See how the method `configureServerCredentials()` is used in for other configurations.

**Review by jeremylong — CHANGES_REQUESTED** [review]
Any place we are accepting credentials we should allow this to be retrieved from the settings.xml.

File: maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java (hunk 1 above)
Comment (mischa-n) [reply maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java:2205]: We have a slightly different setup: the settings.xml is centrally specified, so it is used by all projects. In some places there are actually credentials, but these only refer to environment variables, e.g. `<password>${env.xyz}</password>`. `xyz` is then stored (encrypted) in the CI system.

So in our case, the password would not be stored in plain text in the POM either, but would also refer to an environment variable.


But I will try to implement the requirement anyway.

Comment (mischa-n) [reply maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java:2205]: Should be solved now.

**Comment by mischa-n** [comment]
@jeremylong : is there anything else I can do here to get the PR accepted?

**Review by jeremylong — APPROVED** [review]

**Comment by jeremylong** [comment]
Sorry about the delay - thanks for the PR!

**Pull request merged**